ludwig/api.py

```diff
@@ -580,9 +580,15 @@ def on_epoch_end(self, trainer, progress_tracker, save_path):
 
                 self.training_set_metadata = training_set_metadata
 
-                if not skip_save_model:
-                    # Load the best weights from saved checkpoint
-                    self.load_weights(model_dir)
+                # Ensure model weights are saved to the driver if training was done remotely
+                if self.backend.is_coordinator() and not skip_save_model:
+                    weights_save_path = os.path.join(model_dir, MODEL_WEIGHTS_FILE_NAME)
+                    if not path_exists(weights_save_path):
```

Comment:
Would the file not need to be overwritten as training progresses?

Reply:
This is happening after training has already completed.